rpc_jobs/unit/re_unit_tests.yml

```diff
@@ -22,8 +22,27 @@
       if(env.ghprbPullId != null){
         env.RPC_GATING_BRANCH="origin/pr/${env.ghprbPullId}/merge"
       }
+
+      // We execute this here so that when a PR is submitted
+      // all the parallel builds below do not have to do their
+      // own request for the venv to be built. This helps reduce
+      // the load on jenkins and gets the job done faster.
+      stage("Build-Gating-Venv") {
+        build(
+          job: "Build-Gating-Venv",
+          wait: true,
+          parameters: [
+            [
+              $class: 'StringParameterValue',
+              name: 'RPC_GATING_BRANCH',
+              value: env.RPC_GATING_BRANCH
+            ]
+          ]
+        ) // build
+      } // stage
+
       parallel ([
-        "With Requested Credentials": {
+        stage("With Requested Credentials") {
```

Comment:
Have you rebuilt the job and tested these changes? The examples I've seen have parallel take a map which would no longer be the case if this is done.

Reply:
@git-harry makes a good point; parallel takes a map, of string -> closure. Each parallel branch is effectively a stage so theres little value in adding the stage call. Also `stage()` doesn't return a closure so this wouldn't work even if it was a map. I recommend leaving it how it was. 

The docs are confusing on this, as [declarative pipeline](https://jenkins.io/doc/book/pipeline/syntax/#parallel-stages-example) uses 
```
pipeline {
  stages {
    stage('a'){
      ...
    }
    stage('b'){
      ...
    }
  }
}
```

We currently use scripted pipeline which uses the map syntax - Its easy to get the two syntaxes confused.

Reply:
Alright, will revert it. I primarily did this because it looks prettier in the UI. :p